include/tvm/runtime/memory_manager.h

```diff
@@ -130,7 +131,7 @@ class MemoryManager {
 class StorageObj : public Object {
  public:
   /*! \brief The index into the VM function table. */
-  Buffer buffer;
+  MBuffer buffer;
 
   /*! \brief Allocate an NDArray from a given piece of storage. */
   NDArray AllocNDArray(size_t offset, std::vector<int64_t> shape, DLDataType dtype);
```

Comment:
It would be good to update vector<int_64_t> shape to `ShapeTuple`. I can work on this once this pr lands

Reply:
Handled ShapeTuple across memory_manager and Allocators